physcraper/__init__.py

```diff
@@ -2390,156 +2261,91 @@ def remove_identical_seqs(self):
-                        debug("seq too short")
-                        # do not add them to not added, as seq len depends on sequence which was blasted and
-                        # there might be a better matching seq (one which will return a longer sequence...)
-                        # if gb_id not in self.gb_not_added:
-                        #     self.gb_not_added.append(gb_id)
+                        self.newseqs_acc.append(gb_id)
+                        otu_id = self.data.add_otu(gb_id, self.ids)
+                        self.seq_dict_build(seq, otu_id, tmp_dict)
+                else:
+                    if gb_id not in self.gb_not_added:
+                        self.gb_not_added.append(gb_id)
                         # writeinfofiles.write_not_added_info(self, gb_id, "seqlen_threshold_not_passed")
                         len_seq = len(seq.replace("-", "").replace("N", ""))
-                        reason = "seqlen_threshold_not_passed: len seq: {}, min len: {}".format(len_seq, seq_len_cutoff)
+                        reason = "seqlen_threshold_not_passed: min len: {}, len: {}".format(len_seq, seq_len_cutoff)
                         if "sscinames" in self.data.gb_dict[gb_id]:
                             tax_name = self.data.gb_dict[gb_id]['sscinames']
                             ncbi_id = self.data.gb_dict[gb_id]['staxids']
                         else:
                             tax_name = None
                             ncbi_id = None
                         writeinfofiles.write_not_added(ncbi_id, tax_name, gb_id, reason, self.workdir)
-                        self.gb_not_added.append(gb_id)
-                        # needs to be deleted from gb_dict. If we later blast a seq which fits better with this one,
-                        # it will not be tried to add, as we tried before with a bad matching one
-                        del self.data.gb_dict[gb_id]
+
                         # fn = open("{}/not_added_seq.csv".format(self.workdir), "a+")
                         # fn.write(
                         #     "seqlen_threshold_not_passed, {}, {}, min len: {}\n".format(gb_id, len_seq, seq_len_cutoff))
                         # fn.close()
-        # this assert got more complicated, as sometimes superseqs are already deleted in seq_dict_build() ->
-        # then subset assert it not True
         old_seqs_ids = set()
         for tax in old_seqs:
             old_seqs_ids.add(tax)
-        tmp_dict_plus_super = set()
-        for item in tmp_dict.keys():
-            tmp_dict_plus_super.add(item)
-        for item in self.del_superseq:
-            tmp_dict_plus_super.add(item)
-        # print(tmp_dict_plus_super)
-        # assert old_seqs_ids.issubset(tmp_dict.keys()), ([x for x in old_seqs_ids if x not in tmp_dict.keys()])
-        assert old_seqs_ids.issubset(tmp_dict_plus_super), ([x for x in old_seqs_ids if x not in tmp_dict_plus_super])
+        assert old_seqs_ids.issubset(tmp_dict.keys())
```

Comment:
Please don't remove assert statements I fixed. If something is unclear please ask.

Reply:
This was an accidental removal in the merge, due to this function being edited in both branches - feel free to add any asserts back into the new branch!